go-controller/pkg/ovn/controller/services/repair.go

```diff
@@ -0,0 +1,115 @@
+package services
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/gateway"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/loadbalancer"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+)
+
+// Repair is a controller loop that can work as in one-shot or periodic mode
+// it checks the OVN Load Balancers and delete the ones that doesn't exist in
+// the Kubernetes serviceTracker.
+// Based on:
+// https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.19/pkg/registry/core/service/ipallocator/controller/repair.go
+type Repair struct {
```

Comment:
I'll voice my concern here again about adding a endpoint slice-services reconciliation loop, rather than a generic reconciliation library that all controllers could use. Additionally this does not follow the design pattern of other controllers, and would have liked this type of new design to be discussed as part of a more generic solution to the project rather than an implementation specific to endpoint slices.I guess we can fix this later.

Reply:
I think that there is a miscommunication here, we can/should go through this later and ... but this is not a reconciliation loop :smile: